src/test/java/uk/gov/justice/probation/courtcasematcher/service/MatcherServiceTest.java

```diff
@@ -0,0 +1,244 @@
+        verify(caseMapper, never()).merge(any(Case.class), eq(courtCase));
+        verify(caseMapper, never()).newFromCaseAndOffender(incomingCase, offender);
+        verify(courtCaseRestClient, timeout(REST_CLIENT_WAIT_MS)).putCourtCase(eq(COURT_CODE), eq(CASE_NO), eq(courtCase));
+    }
+
+    @Test
+    void givenIncomingCaseDoesNotMatchExistingCase_andItHasOnePartialMatch_whenMatchCalled_thenCreateANewRecordWithoutOffenderData(){
+        when(courtCaseRestClient.getCourtCase(COURT_CODE, CASE_NO)).thenReturn(Mono.empty());
+        when(offenderSearchRestClient.search(DEF_NAME, DEF_DOB)).thenReturn(Mono.just(singleFuzzyMatch));
+        when(caseMapper.newFromCase(incomingCase)).thenReturn(courtCase);
+        when(courtCaseRestClient.putCourtCase(eq(COURT_CODE), eq(CASE_NO), eq(courtCase))).thenReturn(mock(Disposable.class));
+
+        matcherService.match(incomingCase);
+
+        verify(caseMapper).newFromCase(incomingCase);
+        verify(caseMapper, never()).merge(any(Case.class), eq(courtCase));
+        verify(caseMapper, never()).newFromCaseAndOffender(incomingCase, offender);
+        verify(courtCaseRestClient, timeout(REST_CLIENT_WAIT_MS)).putCourtCase(eq(COURT_CODE), eq(CASE_NO), eq(courtCase));
+    }
+
+    @Test
+    public void whenMatchesReturned_thenLogTheDetails() {
+        when(courtCaseRestClient.getCourtCase(COURT_CODE, CASE_NO)).thenReturn(Mono.empty());
+        when(offenderSearchRestClient.search(DEF_NAME, DEF_DOB)).thenReturn(Mono.just(singleExactMatch));
+        when(caseMapper.newFromCaseAndOffender(incomingCase, offender)).thenReturn(courtCase);
+        when(courtCaseRestClient.putCourtCase(eq(COURT_CODE), eq(CASE_NO), eq(courtCase))).thenReturn(mock(Disposable.class));
+
+        matcherService.match(incomingCase);
+
+        LoggingEvent loggingEvent = captureLogEvent();
+        assertThat(loggingEvent.getLevel()).isEqualTo(Level.INFO);
+        assertThat(loggingEvent.getFormattedMessage().trim())
+                .contains("Match results for caseNo: 1600032952, courtCode: SHF - matchedBy: ALL_SUPPLIED, matchCount: 1");
+    }
+
+    @Test
+    void whenEmptyMonoReturned_thenLogDetails(){
+        when(courtCaseRestClient.getCourtCase(COURT_CODE, CASE_NO)).thenReturn(Mono.empty());
+        when(offenderSearchRestClient.search(DEF_NAME, DEF_DOB)).thenReturn(Mono.empty());
+        when(caseMapper.newFromCase(incomingCase)).thenReturn(courtCase);
+        when(courtCaseRestClient.putCourtCase(eq(COURT_CODE), eq(CASE_NO), eq(courtCase))).thenReturn(mock(Disposable.class));
+
+        matcherService.match(incomingCase);
+
+        LoggingEvent loggingEvent = captureLogEvent();
+        assertThat(loggingEvent.getLevel()).isEqualTo(Level.ERROR);
+        assertThat(loggingEvent.getFormattedMessage().trim())
+                .contains("Match results for caseNo: 1600032952, courtCode: SHF - Empty response from OffenderSearchRestClient");
+    }
+
+    private LoggingEvent captureLogEvent() {
```

Comment:
Great. We might want to use this around other tests. I added for checking log events for event bus related events and we may want more of it for critical log statements that app insights or others might want. 
```suggestion
    private LoggingEvent captureLogEvent() {
```